### ISSUE TITLE
The Future of Powerful APIs on the Web Platform

### ISSUE THREAD

**Issue by cynthia** [w3c/tpac2023-breakouts#76]
### Session description

Powerful APIs have been a constant source of contention when balancing the needs of making the web platform competitive while still providing the fundamental guarantees such as security and privacy. Given the trend of web-platform derivative application runtimes such as Electron becoming popular, the potential and user needs of this is clear - but how we can deliver such functionality while providing the security and privacy guarantees of a browser remains an unsolved problem. 

As a simple example of this being a problem - fundamental platform functionality, such as accessing files is still in an unsatisfying state, leaving the web platform as a less desirable platform for developers to target serious applications against. Is there a path where we can have a standardized replacement for Electron, and if so, what kind of fundamental architectural changes would that entail? 

This session would like to invite members of the community who are interested in contributing towards a solution to this. Of the participants, we would like to create a task force to identify unmet user needs on the web platform, attempt to find a path forward to tackle this problem space, and  compile a set of principles and requirements that solutions developed to address these needs should follow. This venture will function as a TAG-endorsed task force, and the participants will be invited by the TAG chairs. Members may include current or former members of the TAG.

The task force  would be identifying ways to bridge the gap between the Web Platform and native application platforms, while protecting security and user privacy, aligning with the TAG's Ethical Principles, Design Principles, the in-development Privacy Principles and the Privacy & Security Questionnaire. We would need to ensure this effort has multi-stakeholder interest, while ensuring cross-browser and platform support.

### Session goal

Solicit feedback on problem space, iterate on focus area for task force, and recruit participants.

### Additional session chairs (Optional)

Lea Verou (@leaverou), Dan Appelquist (@torgo)

### IRC channel (Optional)

#scary-apis

### Who can attend

Anyone may attend (Default)

### Session duration

60 minutes (Default)

### Estimated number of in-person attendees

Don't know (Default)

### Agenda, minutes, slides, etc.

- Agenda: 10 minutes intro, 10 minutes for problem space definition, 40 minutes for discussion.
- Slides: To be uploaded.
- Minutes: To be linked.

**Comment by ianbjacobs** [comment]
Moving some info here for our formatting purposes:

- Agenda: 10 minutes intro, 10 minutes for problem space definition, 40 minutes for discussion.
- Slides: To be uploaded.
- Minutes: To be linked.

Later you can add links where the @@'s are (if you'd like)

Thanks!

**Comment by scheib** [comment]
This is a good proposal for web app capabilities. 
Also note the breakout session #40 specifically for Installable Web Apps

**Comment by scheib** [comment]
[Scheduling](https://www.w3.org/calendar/tpac2023/breakout-sessions/) currently has a conflict between [Installing Web Apps](https://www.w3.org/events/meetings/c4db57e5-99e9-496d-9b4e-87dd5c421966/) and [The Future of Powerful APIs on the Web Platform
](https://www.w3.org/events/meetings/6d209c82-549d-4506-bc36-764ac4326b14/).

**Comment by ianbjacobs** [comment]
Hi @scheib, can you say more about the source of the scheduling conflict?

**Comment by LeaVerou** [comment]
> Hi @scheib, can you say more about the source of the scheduling conflict?

Hm it appears @cynthia forgot to include the list of conflicts from our draft, here it is:  #27, #33, #40, #42, #56 #58, #60,

**Comment by torgo** [comment]
Hmm.... @scheib @ianbjacobs that doesn't work - the people who would attend installable web apps would be the same people who would want to attend this session. ALSO: I need to be in both #27 and #58 and this one, so that also doesn't work...

**Comment by tidoust** [comment]
> Hm it appears @cynthia forgot to include the list of conflicts from our draft, here it is: https://github.com/w3c/tpac2023-breakouts/issues/27, https://github.com/w3c/tpac2023-breakouts/issues/33, https://github.com/w3c/tpac2023-breakouts/issues/40, https://github.com/w3c/tpac2023-breakouts/issues/42, https://github.com/w3c/tpac2023-breakouts/issues/56 https://github.com/w3c/tpac2023-breakouts/issues/58, https://github.com/w3c/tpac2023-breakouts/issues/60,

Thanks for the list. Now added as explicit conflicts in the description of the session. We'll see how we can adjust the draft schedule to resolve them.

**Comment by dmurph** [comment]
Can we put a link to the notes document / minutes in here?

**Comment by ianbjacobs** [comment]
@dmurph, yes, please edit the session description and replace an @@ with a link to the notes. Thanks!

**Comment by dmurph** [comment]
Oh, I don't have a link, I'm trying to find them 🙃

**Comment by tomayac** [comment]
I have this link to the [slides](https://docs.google.com/presentation/d/1-1Q-2206wTjWadU0NtXozEWyGZbnaz1ABHCWL_MCNaM/edit#slide=id.p).

**Comment by scheib** [comment]
[Minutes](https://docs.google.com/document/d/1vRO6xher_AiHDOkQIClgz4wCI8tEAJdHifLU-M2epOI/edit#heading=h.xxa48zxk3cpj)
